web/src/ui.rs

```diff
@@ -56,9 +91,28 @@ fn get_extension_from_filename(filename: &str) -> Option<String> {
         .map(|x| ".".to_owned() + x)
 }
 
+fn download_as_file(filename: Option<&str>, data: &[u8]) -> Result<(), JsValue> {
+    let array = Uint8Array::from(data);
+    let blob = Blob::new_with_u8_array_sequence(&Array::of1(&array))?;
+    let window = web_sys::window().ok_or(JsValue::from("no window"))?;
+    let document = window.document().ok_or(JsValue::from("no document"))?;
+    let a = document.create_element("a")?;
+
+    let url = JsUrl::create_object_url_with_blob(&blob)?;
+    a.set_attribute("href", &url)?;
+    if let Some(filename) = filename {
+        a.set_attribute("download", filename)?;
```

Comment:
Pretty sure you want to always set this attribute to force a download.

Reply:
That's right, without it the downloaded file may be displayed, fixed